src/CommunityToolkit.Maui/Alerts/Snackbar/Snackbar.shared.cs

```diff
@@ -150,27 +150,7 @@ protected virtual ValueTask DisposeAsyncCore()
 	static TimeSpan GetDefaultTimeSpan() => TimeSpan.FromSeconds(3);
 
 #if ANDROID || IOS || MACCATALYST || WINDOWS
-
-	static NativeSnackbar? nativeSnackbar;
-
-	static NativeSnackbar? NativeSnackbar
-	{
-		get
-		{
-			return MainThread.IsMainThread
-				? nativeSnackbar
-				: throw new InvalidOperationException($"{nameof(nativeSnackbar)} can only be called from the Main Thread");
-		}
-		set
-		{
-			if (!MainThread.IsMainThread)
-			{
-				throw new InvalidOperationException($"{nameof(nativeSnackbar)} can only be called from the Main Thread");
-			}
-
-			nativeSnackbar = value;
-		}
-	}
```

Comment:
This code doesn't run on `net6.0`, only on `net6.0-ios`, `net6.0-android`, `net6.0-maccatalyst`, and `net6.0-windows`, so it shouldn't impact our Unit Tests.

Reply:
MainThread as I left a comment is going to be removed. Also according to implementation Dispatcher checks if it needs to be dispatched and run the code on platform MainThread.
Because our methods Show, Dismiss, Dispose use Dispatcher (and it's the only one way to show Snackbar/Toast) from my point of view this check is redundant.

Reply:
Yup, you're right that it's totally redundant because `Show()` and `Dismiss()` always push to the UI thread. 

I added the redundancy on purpose for two reasons:
- If `Show()` and `Dismiss()` are ever modified to remove the Dispatcher, we will still validate that we're using the correct thread
- Since `Snackbar` is a partial class, if anything in `Snackbar.[platorm].cs` references `Snackbar`, this ensures `NativeSnackbar` is always referenced from the UI thread

All that said, this isn't really something I care enough about to hold up this PR! If you'd like to keep it out, that's cool. Just let me know and I'll approve this PR 👍

Reply:
Fair enough. We can refactor `Show()` and `Dismiss()` to remove the marshaling to the UI thread.

I strongly disagree that a library should provide code that throws a preventable exception, but I'm happy to be outvoted by Vlad, Pedro and Shaun 🙌